sos/cleaner/obfuscation_archive.py

```diff
@@ -0,0 +1,222 @@
+        self.build_tar_file()
+        exec_cmd = "%s %s" % (cmd, self.tarpath)
+        res = sos_get_command_output(exec_cmd, timeout=0, stderr=True)
+        if res['status'] == 0:
+            self.final_archive_path = self.tarpath + '.' + exec_cmd[0:2]
+            self.log_debug("Compressed to %s" % self.final_archive_path)
+            try:
+                self.remove_extracted_path()
+            except Exception as err:
+                self.log_debug("Failed to remove extraction directory: %s"
+                               % err)
+                self.report_msg('Failed to remove temporary extraction '
+                                'directory')
+        else:
+            err = res['output'].split(':')[-1]
+            self.log_debug("Exception while compressing archive: %s" % err)
+            raise Exception(err)
+
+    def remove_extracted_path(self):
+        """After the tarball has been re-compressed, remove the extracted path
+        so that we don't take up that duplicate space any longer during
+        execution
+        """
+        def force_delete_file(action, name, exc):
+            os.chmod(name, stat.S_IWUSR)
+            if os.path.isfile(name):
+                os.remove(name)
+            else:
+                shutil.rmtree(name)
+        if os.getuid() != 0:
+            # this is rather silly, but due to the fact we preserve permissions
+            # in reports, a non-root user would be unable to remove a lot of
+            # paths under proc/ and sys/. Since we're at the point where the
+            # archive has been re-compressed and we're removing data that will
+            # no longer be referenced anywhere, just open the perms to allow
+            # us to delete it all
+            sos_get_command_output("chmod -R 777 %s" % self.extracted_path)
+        self.log_debug("Removing %s" % self.extracted_path)
+        shutil.rmtree(self.extracted_path, onerror=force_delete_file)
+
+    def extract_self(self):
+        """Extract an archive into our tmpdir so that we may inspect it or
+        iterate through its contents for obfuscation
+        """
+        archive = tarfile.open(self.archive_path)
+        path = os.path.join(self.tmpdir, 'cleaner')
+        archive.extractall(path)
+        archive.close()
+        return os.path.join(path, archive.name.split('/')[-1].split('.tar')[0])
+
+    def get_file_list(self):
+        """Return a list of all files within the archive"""
+        self.file_list = []
+        for dirname, dirs, files in os.walk(self.extracted_path):
+            for filename in files:
+                self.file_list.append(os.path.join(dirname, filename))
+        return self.file_list
+
+    def update_sub_count(self, fname, count):
+        """Called when a file has finished being parsed and used to track
+        total substitutions made and number of files that had changes made
+        """
+        self.file_sub_list.append(fname)
+        self.total_sub_count += count
```

Comment:
`total_sub_count` is currently never used. But makes sense to track stats, for sure.

Reply:
Right, prepping this for metadata inclusion.